### ISSUE TITLE
Provide Tensor Padding Helpers

### ISSUE THREAD

**Issue by dcvz** [tracel-ai/burn#960]
### Feature description

Would be great to have helpers in Tensor for padding. Similar to PyTorch's [pad](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) and Candle's [pad_with_zeros](https://github.com/huggingface/candle/blob/c6763e3b41790bea0459d3838c3ed1d59d5174ef/candle-core/src/tensor.rs#L2246)

### Additional Information

Some API like this would likely be good (result of conversation on Discord):

```rust
pub fn pad<const D: usize>(self, pad: Padding, pad_mode: PadMode, value: K::Elem) -> Tensor<B, D, K>


// Possible definition of padding
pub struct Padding {
    left: usize,
    right: usize,
    top: usize,
    bottom: usize,
}
impl Padding {
    pub fn uniform(value: usize) -> Self {
        Self {
            left: value,
            right: value,
            top: value,
            bottom: value,
        }
    }
    
    // Additional helpers to create useful padding configs
}
```

**Comment by jcmullwh** [comment]
Hi, I can take a shot at this one!
